src/proto/streams/streams.rs

```diff
@@ -687,18 +709,29 @@ where
 
         let me = &mut *me;
 
-        let id = {
-            let mut stream = me.store.resolve(self.key);
-            stream.ref_dec();
-
-            if !stream.is_released() {
-                return;
-            }
-
-            stream.remove()
-        };
+        let mut stream = me.store.resolve(self.key);
+        // decrement the stream's ref count by 1.
+        stream.ref_dec();
 
-        debug_assert!(!me.store.contains_id(&id));
```

Comment:
Why remove this assertion?

Reply:
i believe this was originally removed due to ownership reasons - it may no longer be necessary to remove, I can double check.

Reply:
@seanmonstar the assertion also doesn't make sense here any longer, because whether or not the stream needs to be removed from the store is now handled by `transition`.